test/libsolidity/syntaxTests/specialFunctions/types_with_unspecified_encoding.sol

```diff
@@ -0,0 +1,31 @@
+contract C {
+    struct S { uint x; }
+    S s;
+    struct T { }
+    T t;
+    enum A { X, Y }
+    function f() public pure {
+        bytes32 h = keccak256(keccak256, f, this.f.gas, block.blockhash);
+        bool a = address(this).call(address(this).delegatecall, super);
+        bool b = address(this).delegatecall(log0, tx, mulmod, S, A, T, uint, uint[]);
+        bytes32 c = sha256(s, t);
+        h; a; b; c;
+    }
+}
+// ----
+// TypeError: This type cannot be encoded.
```

Comment:
While the test probably is good, because the number of errors reflects the number of unsupported types, it would be nice to split this into multiple tests. At least structs could be moved out.

Reply:
Will do!

Reply:
Done.